python/ray/experimental/dag/class_node.py

```diff
@@ -1,37 +1,60 @@
 import ray
 from ray.experimental.dag.dag_node import DAGNode
+from ray.experimental.dag.input_node import InputNode
 
 from typing import Any, Dict, List, Optional, Tuple
 
 
 class ClassNode(DAGNode):
     """Represents an actor creation in a Ray task DAG."""
 
-    def __init__(self, cls: type, cls_args, cls_kwargs, cls_options):
+    def __init__(
+        self,
+        cls,
+        cls_args,
+        cls_kwargs,
+        cls_options,
+        other_args_to_resolve=None,
+    ):
         self._body = cls
         self._last_call: Optional["ClassMethodNode"] = None
-        DAGNode.__init__(self, cls_args, cls_kwargs, cls_options)
+        super().__init__(
+            cls_args,
+            cls_kwargs,
+            cls_options,
+            other_args_to_resolve=other_args_to_resolve,
+        )
+
+        if self._contain_input_node():
+            raise ValueError(
```

Comment:
"InputNodes cannot be used as arguments to actor constructors, only task and actor method calls."